### PR TITLE
NOTIF-190 Add gauge to visualize aggregated mails in grafana

### PR DESCRIPTION
**Pull request by bennetelli**

### CODE REVIEW

File: aggregator/pom.xml [hunk 1]
```diff
@@ -42,6 +42,14 @@
...
             <artifactId>quarkus-smallrye-reactive-messaging-kafka</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.micrometer</groupId>
+            <artifactId>micrometer-registry-prometheus</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-micrometer</artifactId>
+        </dependency>
```
Comment (gwenneg) [inline aggregator/pom.xml:52]:
```suggestion
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-micrometer-registry-prometheus</artifactId>
        </dependency>
```

File: aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java [hunk 2]
```diff
@@ -136,7 +138,7 @@ void shouldNotIncreaseAggregationsWhenHostIdIsDifferent() {
     void shouldNotProcessMailsWhenNewCronJobIsDisabledByDefault() {
         final EmailAggregationResources emailAggregationResources = mock(EmailAggregationResources.class);
         final ObjectMapper objectMapper = mock(ObjectMapper.class);
-        DailyEmailAggregationJob dailyEmailAggregationJob = new DailyEmailAggregationJob(emailAggregationResources, objectMapper);
+        DailyEmailAggregationJob dailyEmailAggregationJob = new DailyEmailAggregationJob(emailAggregationResources, objectMapper, new SimpleMeterRegistry());
```
Comment (gwenneg) [inline aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java:141]: Shouldn't the counter value be checked in all tests?

File: aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java [hunk 3]
```diff
@@ -44,6 +45,7 @@
     void setUp() {
         helpers.purgeEmailAggregations();
         System.clearProperty("notifications.aggregator.email.subscription.periodic.cron.enabled");
+        testee.registry = new SimpleMeterRegistry();
```
Comment (gwenneg) [inline aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java:48]: Note for later:
Injected fields should not be instantiated with `new` in a `QuarkusTest`, they should be mocked instead.

**Review by gwenneg — CHANGES_REQUESTED** [review]

File: aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java (hunk 2 above)
Comment (bennetelli) [reply aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java:141]: I actually don't want to test the gauge in that state. Will probably have to add another gauge to visualize more things and will then aggregate all gauges containing to the cron job in the same custom gauge. This is just a first step.

File: aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java (hunk 3 above)
Comment (bennetelli) [reply aggregator/src/test/java/com/redhat/cloud/notifications/DailyEmailAggregationJobTest.java:48]: Yes, this test is also a bit misleading as it's testing with mocks, as asll as integration testing without mocks.
I think in the long term it's better to split the test.

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**